File: src/events/guildMemberAdd.js
const { Events } = require('discord.js');
const config = require('../../config/config.js');
const settings = require('../../config/settings.json');

module.exports = {
    name: Events.GuildMemberAdd,
    async execute(member) {
        // Solo actualizar si es el servidor configurado
        if (member.guild.id === settings.guildId && !member.user.bot) {
            member.client.updatePresenceCount({ delta: 1 }).catch(() => {});
        }
        if (config.autoroles.enabled && config.autoroles.roles.length > 0) {
            try {
                const roles = config.autoroles.roles
                    .map(roleId => member.guild.roles.cache.get(roleId))
                    .filter(role => role != null);

                if (roles.length > 0) {
                    await member.roles.add(roles);
                }
            } catch (error) {
                console.error('Error al asignar autoroles:', error);
            }
        }
    },
};

File: pybot/events/guild_member_add.py
from __future__ import annotations

from discord.ext import commands

from .. import config


class GuildMemberAddEvents(commands.Cog):
    def __init__(self, bot: commands.Bot) -> None:
        self.bot = bot

    @commands.Cog.listener()
    async def on_member_join(self, member):
        settings = config.resolve_config()
        if member.guild and settings.get("guildId") and str(member.guild.id) == str(settings.get("guildId")):
            if not member.bot:
                await self.bot.update_presence_count(delta=1)

        autoroles = settings.get("autoroles", {})
        if autoroles.get("enabled") and autoroles.get("roles"):
            try:
                roles = [member.guild.get_role(int(role_id)) for role_id in autoroles.get("roles", [])]
                roles = [role for role in roles if role]
                if roles:
                    await member.add_roles(*roles)
            except Exception:
                pass


async def setup(bot: commands.Bot) -> None:
    await bot.add_cog(GuildMemberAddEvents(bot))
